pkg/manifest/firstboot.go

```diff
@@ -0,0 +1,168 @@
+package manifest
+
+import (
+	"fmt"
+	"html/template"
+	"io/fs"
+	"strings"
+
+	"github.com/osbuild/images/internal/common"
+	"github.com/osbuild/images/pkg/customizations/firstboot"
+	"github.com/osbuild/images/pkg/customizations/fsnode"
+	"github.com/osbuild/images/pkg/osbuild"
+)
+
+// checkName prevents path traversal
+func checkName(str string) error {
+	if str == "" {
+		return fmt.Errorf("name cannot be empty")
+	}
+
+	for _, r := range str {
+		if !(('a' <= r && r <= 'z') || ('A' <= r && r <= 'Z') || ('0' <= r && r <= '9') || r == '-' || r == '_') {
+			return fmt.Errorf("name can only contain alphanumeric characters, dashes, and underscores")
+		}
+	}
+
+	return nil
+}
+
+var tmplFirstbootAAP = `#!/usr/bin/bash
+curl -s -i --data "host_config_key={{ .HostConfigKey }}" {{ .URL -}}
```

Comment:
And if we are being defensive then we might as well escape this too so it's obvious and local. We have `Quote` in `pkg/util/shlex_quote.go`.